examples/name-service/programs/name-service/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+use anchor_lang::prelude::*;
+use light_hasher::{errors::HasherError, DataHasher, Hasher, Poseidon};
+use light_system_program::{
+    invoke::processor::CompressedProof,
+    sdk::{
+        address::derive_address,
+        compressed_account::{
+            CompressedAccount, CompressedAccountData, PackedCompressedAccountWithMerkleContext,
+        },
+        CompressedCpiContext,
+    },
+    InstructionDataInvokeCpi, NewAddressParamsPacked, OutputCompressedAccountWithPackedContext,
+};
+use light_utils::hash_to_bn254_field_size_be;
+
+declare_id!("7yucc7fL3JGbyMwg4neUaenNSdySS39hbAk89Ao3t1Hz");
+
+pub const DISCRIMINATOR: &[u8; 8] = b"name-ser";
+
+#[program]
+pub mod name_service {
+
+    use super::*;
+
+    pub fn create_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, NameService<'info>>,
+        proof: CompressedProof,
+        new_address_params: NewAddressParamsPacked,
+        name: String,
+        rdata: RData,
+        cpi_context: Option<CompressedCpiContext>,
+    ) -> Result<()> {
+        let record = NameRecord {
+            owner: ctx.accounts.signer.key(),
+            name,
+            rdata,
+        };
+        let compressed_account = create_compressed_account(&ctx, &record, &new_address_params)?;
+
+        let signer_seed = b"cpi_signer".as_slice();
+        let bump = Pubkey::find_program_address(&[signer_seed], &ctx.accounts.self_program.key()).1;
+        let signer_seeds: [&[u8]; 2] = [signer_seed, &[bump]];
+
+        let inputs = InstructionDataInvokeCpi {
+            proof: Some(proof),
+            new_address_params: vec![new_address_params],
+            input_compressed_accounts_with_merkle_context: Vec::new(),
+            output_compressed_accounts: vec![compressed_account],
+            relay_fee: None,
+            compression_lamports: None,
+            is_compress: false,
+            signer_seeds: signer_seeds.iter().map(|seed| seed.to_vec()).collect(),
+            cpi_context,
+        };
+        let inputs = inputs.try_to_vec()?;
+
+        let signer_seeds: [&[&[u8]]; 1] = [&signer_seeds[..]];
+        let mut cpi_ctx = CpiContext::new_with_signer(
+            ctx.accounts.light_system_program.to_account_info(),
+            cpi_accounts(&ctx),
+            &signer_seeds[..],
+        );
+
+        cpi_ctx.remaining_accounts = ctx.remaining_accounts.to_vec();
+
+        light_system_program::cpi::invoke_cpi(cpi_ctx, inputs)?;
```

Comment:
can you try these from `light-sdk`:
`create_cpi_inputs_for_new_address` and `verify`
and let me know whether they work for you, what would need to change, what's unintuitive, etc?

Reply:
Nice one! I like both of the helpers.

I think that we also need some helper(s) similar to `create_cpi_inputs_for_new_address`, but for modyfing the account - where you would pass:

- old account to nullify
- new account
- already existing address